content/en/docs/setup/certificates.md

```diff
@@ -0,0 +1,136 @@
+You can also generate your own certificates -- for example, to keep your private keys more secure by not storing them on the API server.
+This page explains the certificates that your cluster requires.
+
+{{% /capture %}}
+
+{{% capture body %}}
+
+## How certificates are used by your cluster
+
+Kubernetes requires PKI for the following operations:
+
+* Client certificates for the kubelet to authenticate to the API server
+* Server certificate for the API server endpoint
+* Client certificates for administrators of the cluster to authenticate to the API server
+* Client certificates for the API server to talk to the kubelets
+* Client certificate for the API server to talk to etcd
+* Client certificate/kubeconfig for the controller manager to talk to the API server
+* Client certificate/kubeconfig for the scheduler to talk to the API server.
+* Client and server certificates for the [front-proxy][proxy]
+
+etcd also implements mutual TLS to authenticate clients and peers.
+
+## Where certificates are stored
+
+Kubernetes stores certificates by default in `/etc/kubernetes/pki`. All paths in this documentation are relative to that directory.
+
+## Configure certificates manually
+
+If you don't want [kubeadm][kubeadm] to generate the required certificates, you can create them in either of the following ways.
+
+### Single root CA
+
+You can create a single root CA, controlled by an adminstrator. This root CA can then create multiple intermediate CAs, and delegate all further creation to Kubernetes itself. 
+
+Required CAs:
+
+| path                   | Default CN                | description                      |
+|------------------------|---------------------------|----------------------------------|
+| ca.crt,key             | kubernetes-ca             | Kubernetes general CA            |
+| etcd/ca.crt,key        | etcd-ca                   | For all etcd-related functions   |
+| front-proxy-ca.crt,key | kubernetes-front-proxy-ca | For the [front-end proxy][proxy] |
+
+### All certificates
+
+If you don't wish to copy these private keys to your API servers, you can generate all certificates yourself. 
+
+Required certificates:
+
+| Default CN                    | Parent CA                 | O (in Subject) | kind                                   | hosts (SAN)                                 |
+|-------------------------------|---------------------------|----------------|----------------------------------------|---------------------------------------------|
+| kube-etcd                     | etcd-ca                   |                | server, client [<sup>1</sup>][etcdbug] | `localhost`, `127.0.0.1`                        |
+| kube-etcd-peer                | etcd-ca                   |                | peer                                   | `<hostname>`, `<Host_IP>`, `localhost`, `127.0.0.1` |
```

Comment:
It would probably be better to standardize both of these to either use `server, client` or `peer`, rather than using both.

Reply:
done